### PR TITLE
WIP: Zowe HA documentation changes

### PR DESCRIPTION
**Pull request by NayerNajafi**
Signed-off-by: Nayer Najafi <nayer.najafi@ibm.com>

### Your checklist for this pull request
🚨Please review the [guidelines for contributing](../docs/contribute/contributing.md) to this repository.

- [X] If the changes in this PR is part of the next future release, make this pull request against the **docs-staging** branch which will be published at the next release boundary. If the changes in this PR are part of the current release, use the default base branch, **master**. For more information about branches, see https://github.com/zowe/docs-site/tree/master#understanding-the-doc-branches. 
      
- [ ] If this PR relates to GitHub issues in `docs-site` or other repositories, please list in Description, prefixed with **close**, **fix** or **resolve** keywords.

### Description (including links to related git issues)
This PR includes the changes for Zowe HA documentation.

:heart:Thank you!

### CODE REVIEW

File: docs/getting-started/overview.md [hunk 1]
```diff
@@ -147,6 +147,7 @@ The API Mediation Layer provides a single point of access for mainframe service
 * Consistent Access: API routing and standardization of API service URLs through the Gateway component provides users with a consistent way to access mainframe APIs at a predefined address.
 * Dynamic Discovery: The Discovery Service automatically determines the location and status of API services.
 * High-Availability: API Mediation Layer is designed with high-availability of services and scalability in mind.
+* Caching Service: This feature is designed for Zowe components in a high availability configuration. It supports the High Availability of all components within Zowe, components either need to be stateless, or offload the state to a location accessible by all instances of the service, including those which just started.
```
Comment (CarsonCook) [inline docs/getting-started/overview.md:150]: Perhaps this instead:
```suggestion
* Caching Service: This feature is designed for Zowe components in a high availability configuration. It supports the High Availability of all components within Zowe, **allowing components to be stateless by providing a mechanism to offload their** state to a location accessible by all instances of the service, including those which just started.
```

**Review by CarsonCook — COMMENTED** [review]
In https://docs.zowe.org/stable/user-guide/configure-instance-directory.html
 There is 
 ```
 High Availability
The high availability (HA) feature of Zowe is under development and has not been fully delivered. The following values are work in progress towards HA capability. They are not used and will be documented in more detail once HA support is finalized in a future Zowe release.

ZWE_DISCOVERY_SERVICES_LIST: (Work in progress) Do not modify this value from its supplied default of https://${ZOWE_EXPLORER_HOST}:${DISCOVERY_PORT}/eureka/.
ZWE_CACHING_SERVICE_PORT=7555: (Work in progress) This port is not yet used so the value does not need to be availale.
ZWE_CACHING_SERVICE_PERSISTENT=VSAM: (Work in progress)
ZWE_CACHING_SERVICE_VSAM_DATASET: (Work in progress)
```
Should updating that section be part of this PR? If so I can draft something.

**Comment by NayerNajafi** [comment]
> In https://docs.zowe.org/stable/user-guide/configure-instance-directory.html
> There is
> 
> ```
> High Availability
> The high availability (HA) feature of Zowe is under development and has not been fully delivered. The following values are work in progress towards HA capability. They are not used and will be documented in more detail once HA support is finalized in a future Zowe release.
> 
> ZWE_DISCOVERY_SERVICES_LIST: (Work in progress) Do not modify this value from its supplied default of https://${ZOWE_EXPLORER_HOST}:${DISCOVERY_PORT}/eureka/.
> ZWE_CACHING_SERVICE_PORT=7555: (Work in progress) This port is not yet used so the value does not need to be availale.
> ZWE_CACHING_SERVICE_PERSISTENT=VSAM: (Work in progress)
> ZWE_CACHING_SERVICE_VSAM_DATASET: (Work in progress)
> ```
> 
> Should updating that section be part of this PR? If so I can draft something.

@CarsonCook  Yes please, it would be great. Thanks

**Comment by jandadav** [comment]
@NayerNajafi updated the apiml diagram and added source for future modifications. In regards to caching service docs, the changes here are enough. We have another doc PR opened for caching service documentation here: https://github.com/zowe/docs-site/pull/1506 which should cover it enough. Please let me know if you have any concern or see any gaps.

**Comment by NayerNajafi** [comment]
> @NayerNajafi updated the apiml diagram and added source for future modifications. In regards to caching service docs, the changes here are enough. We have another doc PR opened for caching service documentation here: #1506 which should cover it enough. Please let me know if you have any concern or see any gaps.

@jandadav Thanks David, the new diagram looks really good.

File: docs/user-guide/install-ha-sysplex.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+   TSS ADD(&ZOWEUSER.) FAC(STC)   
+
+   ```
+
+   Where, 
+   - ZLNCHSTC is the Zowe launcher task name. Default should be ZWESLSTC.
+   - STCGRP is the group for Zowe started tasks. Default should be ZWEADMIN.
+   - ZOWEUSER is the user ID for the Zowe started task. Default should be ZWESVUSR.
+
+2. Create a VSAM data set which is used by the Caching Service feature of API Mediation Layer. For instructions, see []
```
Comment (nannanli) [inline docs/user-guide/install-ha-sysplex.md:126]: Hi @NayerNajafi , is there a specific topic where people could learn how to create this VSAM data set? We should also add a link for the VSAM step in the diagram so wondering which topic we should link to. 
The APIML has an open PR https://github.com/zowe/docs-site/pull/1506/files. Does this contain any information that we could refer to? Thanks!

**Review by nannanli — APPROVED** [review]
I completed the edits. This PR LGTM now. FYI that there is a specific section for HA docs on the site after refactoring the sidebar (see the screen capture below) 
image

I know we're still working on the HA architecture diagram.  Suggest we merge the large piece of docs in now and push late changes to the docs-staging branch later to catch up with the release publish on Jun 14.

**Comment by NayerNajafi** [comment]
> I completed the edits. This PR LGTM now. FYI that there is a specific section for HA docs on the site after refactoring the sidebar (see the screen capture below)
> image
> 
> I know we're still working on the HA architecture diagram. Suggest we merge the large piece of docs in now and push late changes to the docs-staging branch later to catch up with the release publish on Jun 14.

@nannanli Thanks Ashley, looks good. There is another part left for HA documentation as well "[Topology of the Zowe z/OS launch process](https://docs.zowe.org/stable/user-guide/installandconfig.html#topology-of-the-zowe-z-os-launch-process)" that might require some changes.

**Review by jackjia-ibm — APPROVED** [review]

**Pull request merged**